src/main/java/com/schematical/chaoscraft/ai/NeuronBase.java

```diff
@@ -1,24 +1,26 @@
 package com.schematical.chaoscraft.ai;
 
-import com.google.common.collect.Sets;
-import com.schematical.chaoscraft.ChaosCraft;
 import com.schematical.chaosnet.model.ChaosNetException;
 import org.json.simple.JSONArray;
 import org.json.simple.JSONObject;
 
 import java.util.ArrayList;
 import java.util.Iterator;
 import java.util.List;
-import java.util.Set;
 
 /**
  * Created by user1a on 12/8/18.
  */
 public abstract class NeuronBase extends InnovationBase {
 
     public String id;
+
+    //Visualisation stuff
     public int x;
     public int y;
+    public float layer;
+    private int distanceFromOutput;
```

Comment:
I love what you are doing but I almost think this should be some type of GUI Wrapper class. I want to keep the actual Neurons/NNet as trim as possible for CPU / Network sake. Thoughts?

Reply:
Yeah I understand, tho for networking you dont have to sync it (they will only have values client side anyway) and CPU doesnt matter anyway. But I can try to get it out of here